### ISSUE TITLE
Multiple accounts causes sign out on page reload

### ISSUE THREAD

**Issue by wprater** [GoogleWebComponents/google-signin#72]
If one has multiple accounts setup and authenticates with a couple of them, switching back and forth on these accounts causes one of them to never immediately auth after a page refresh.  Only one of the accounts will allow itself to persist auth.

Furthermore, the accounts that don't immediately sign back in call the `handleAuthResult` callback twice each with the `authResult.error` set to `user_signed_out`.

Ive just placed the single element on the page to test this and can't seem to wrap my head around what's happening.  Thoughts?

**Comment by wprater** [comment]
in a separate test that uses a [pure js flow](https://developers.google.com/+/web/signin/javascript-flow), I can get similar behavior when loading `client.js` vs `client:platform.js`.  Overall, it's seems to be quite inconsistent when using multiple accounts.

**Comment by wprater** [comment]
@addyosmani what is the consequence of using `client:platform.js` in `google-client-api`?  It seems to solve the issue.  I've also [created a workaround](http://asdf) to `authenticate` immediately if the `user_signed_out` error is present in the "page-level configuration the callback".  The workaround uses `gapi.auth.authorize` which has the side effect of not working with `signOut`.  So I modified the google-signin `signOut` method, which is not ideal.

My workaround is referenced in this bug and I added a few references that may help to diagnose.

**Comment by Zoramite** [comment]
Can you clarify what you mean by "Switching back and forth on these accounts"?

**Comment by wprater** [comment]
if you're to logout and and log into another account with the google
account switcher.

**Comment by gattasrikanth** [comment]
I had the same issue :( Till now was checking everywhere on internet except here of whats going wrong!!!
I have two accounts signed in the browser.
Open the demo page http://googlewebcomponents.github.io/google-signin/components/google-signin/demo.html
Click Sign in --> Choose account#1
Reload the page --> Account#1 is still signed in.
But If I choose Account#2 this user gets signed out after reloading the page.
I am newbie here but my guess is.. it is only trying to sign in first account by default every time.

**Comment by Zoramite** [comment]
I did some checking. Google+ Sign-in does not support using multiple accounts. So this isn't a limitation specifically with the web component, but with the signin library itself.

**Comment by gattasrikanth** [comment]
Thanks for the update. Can you suggest any workarounds? Anything I can manually edit inside google-signin.html file to check for second account? In my scenario all the users of my webapp are for sure logged with two accounts and I always need that second account only.

**Comment by wprater** [comment]
@gattasrikanth https://github.com/GoogleWebComponents/google-signin/issues/72#issuecomment-73336930 seems to fix it for me, however, Im not sure the consequence, so I never went to use it that way.

**Comment by gattasrikanth** [comment]
the workaround you mentioned says will only work on localhost. so how do implement it. i am not sure how to use platform.js and google-client-api that you mentioned.

**Comment by wprater** [comment]
on older versions you could alter the lib url to `client.js:platform` to
fix the issue.  they've since changed how libs are loaded with the jsapi,
and I have not linked into it yet.

On an older commit, you can see where you'd change the file.  I would not
recommend this approach, since I never got any feedback as to the
repercussion of the change.
https://github.com/GoogleWebComponents/google-apis/blob/4f88265a49ecf9c900fb2eddac6afcccd6d3f1b3/google-client-api.html#L36

**Comment by gattasrikanth** [comment]
Thanks again! I will take a look at it. My application is specific to a google apps domain and for a limited set of users only so i would like to fork and make a workaround. Mostly everything else on my app is finished and am just stuck with this bug. I have also tried if there is any way to include "hd=domain.com" so that only my domain users will be able to sign in but it looks like thats not possible either.
Although hoping for a longterm fix from signin library as well!!

**Comment by gattasrikanth** [comment]
I tried to change to platform.js but it doesn't seems tobe working. (Tested on appspot.com not on localhost). I have also tried to make some changes to google-signin.html like below.
if(result && result.error == 'immediate_failed') {
 var param = {'client_id': clientid, 'cookiepolicy': website, 'scope': scope, 'authuser': '1',
	            'requestvisibleactions': '','immediate': true,
        	}
gapi.auth.authorize(param, handleAuthResult); 

This is still failing and doesn't seem to try signin with authuser=1.
